lib/pbench/test/unit/common/test_config.py

```diff
@@ -29,224 +29,119 @@ def get(self, section: str, option: str, fallback=None) -> str:
+        assert config.log_fmt == log_format
 
-    def test_logger_type_hostport(self, monkeypatch):
-        """verify logger_type value as `hostport`, it expects `host` and `port`"""
+    def test_bad_logger_hostport(self, monkeypatch):
+        """Show that PbenchConfig raises 'BadConfig' and 'ValueError' when
+        the configuration contains no option or an empty string respectively"""
 
-        class mock_MockedConfigParser(MockedConfigParser):
-            def __init__(self):
-                self._sections = {"logging": {"logger_type": "hostport"}}
-
-        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
-        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
-        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        logger_type = "hostport"
+        self.logger_test_setup(monkeypatch, {"logging": {"logger_type": logger_type}})
         with pytest.raises(BadConfig) as exc:
-            config = PbenchConfig("pbench.cfg")
-            assert config.logger_type == "hostport"
-            assert exc == "No option 'logger_host' in section: 'logging'"
-
-    def test_logger_type_host_value(self, monkeypatch):
-        """verify `host` value of logger_type `hostport`"""
-
-        class mock_MockedConfigParser(MockedConfigParser):
-            def __init__(self):
-                self._sections = {
-                    "logging": {"logger_type": "hostport", "logger_host": ""}
-                }
-
-        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
-        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
-        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+            PbenchConfig("pbench.cfg")
+        expected_error_msg = "No option 'logger_host' in section: 'logging'"
+        assert expected_error_msg in str(exc)
+        self.logger_test_setup(
+            monkeypatch, {"logging": {"logger_type": logger_type, "logger_host": ""}}
+        )
         with pytest.raises(ValueError) as exc:
-            config = PbenchConfig("pbench.cfg")
-            assert config.logger_type == "hostport"
-            assert exc == "No option 'logger_host' in section: 'logging'"
-
-    def test_logger_type_port(self, monkeypatch):
-        """verify `logger_port` for logger_type `hostport`"""
-
-        class mock_MockedConfigParser(MockedConfigParser):
-            def __init__(self):
-                self._sections = {
-                    "logging": {"logger_type": "hostport", "logger_host": "localhost"}
-                }
-
-        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
-        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
-        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+            PbenchConfig("pbench.cfg")
+        expected_error_msg = "ValueError('logger_host', 'logging')"
```

Comment:
The sub-cases in this scenario aren't really sharing much in the way of setup.  So, if you aren't going to go with the parameterized approach, then I would split these sub-cases into separate cases.

Having each test exercising a single scenario is helpful...if for no other reason that it will help highlight it when you miss an assertion.  (E.g., at line 102 you set up `expected_error_msg`, but there's no subsequent `assert` which checks it!  😈)

Reply:
I didn't realize that I missed an assertion. What is the parametrized approach?

Reply:
I went with the parametrize approach.. I hope it looks good :D